relay-server/src/actors/envelopes.rs

```diff
@@ -1805,8 +1822,10 @@ impl EnvelopeProcessor {
                 .map_err(ProcessingError::SerializeFailed)?
         });
 
-        let event_type = state.event_type().unwrap_or_default();
-        let mut event_item = Item::new(ItemType::from_event_type(event_type));
+        let mut event_item = match state.event_type().unwrap_or_default() {
+            EventType::Transaction => state.transaction_item.take().unwrap(),
+            ty => Item::new(ItemType::from_event_type(ty)),
+        };
```

Comment:
I assume we reuse the item here because we do not want to lose the `metrics_extracted` information. But this could have side effects (I don't know what else is in that item header). So it might be safer to explicitly transfer the `metrics_extracted` information to the new item, as we do with `sample_rates` a few lines below.